lib/metadata_json_lint.rb

```diff
@@ -83,6 +96,10 @@ def parse(metadata)
       abort("Error: Unable to read metadata file: #{e.exception}")
     end
 
+    abort('Error: metadata.json does not have a valid newline at the end') if contains_invalid_newline?(f)
+
+    abort('Error: Unable to parse metadata.json: Invalid escape character in string') if contains_invalid_escape?(f)
```

Comment:
Thinking about the pattern here: should we display all errors as far as we can?

Also, you can still perfectly parse it in JSON so perhaps it should still run through the schema validations, then show them as regular errors where possible?

Reply:
that requires some more changes. I don't have the time for that at the moment.

Reply:
as discussed we can merge this now and if we've time, improve it later. I documented it at https://github.com/voxpupuli/metadata-json-lint/issues/155